File: OptimoveSDK/app/src/main/java/com/optimove/android/optimovemobilesdk/MyDeferredDeepLinkHandler.java
package com.optimove.android.optimovemobilesdk;


import android.content.Context;
import android.util.Log;

import androidx.annotation.Nullable;

import com.optimove.android.optimobile.DeferredDeepLinkHandlerInterface;
import com.optimove.android.optimobile.DeferredDeepLinkHelper;


public class MyDeferredDeepLinkHandler implements DeferredDeepLinkHandlerInterface {
    private static final String TAG = MyDeferredDeepLinkHandler.class.getName();

    public void handle(Context context, DeferredDeepLinkHelper.DeepLinkResolution resolution, String link, @Nullable DeferredDeepLinkHelper.DeepLink data) {
        //- Inspect the data payload and run code as needed.

        String output = "DDl resolution: " + resolution + " ";
        if (data == null) {
            output += "null data";
        } else {
            output += data.data.toString();
        }
        Log.d(TAG, output);
    }

}

File: OptimoveSDK/app/src/main/java/com/optimove/android/optimovemobilesdk/MyDeferredDeepLinkHandler.kt
package com.optimove.android.optimovemobilesdk

import android.content.Context
import android.util.Log
import android.widget.Toast
import com.optimove.android.optimobile.DeferredDeepLinkHandlerInterface
import com.optimove.android.optimobile.DeferredDeepLinkHelper

class MyDeferredDeepLinkHandler : DeferredDeepLinkHandlerInterface {

    override fun handle(
        context: Context,
        resolution: DeferredDeepLinkHelper.DeepLinkResolution,
        link: String,
        data: DeferredDeepLinkHelper.DeepLink?
    ) {
        val output = "DDl resolution: $resolution ${data?.data?.toString() ?: "null data"}"
        Log.d(TAG, output)
        Toast.makeText(context, output, Toast.LENGTH_SHORT).show()
    }

    companion object {
        private val TAG = MyDeferredDeepLinkHandler::class.java.name
    }
}
